### PR TITLE
feat(cli): use requestedUrl instead of finalUrl for upload context naming

### PR DESCRIPTION
**Pull request by jaspritt-hora**
**CHANGED**:

- Using `requestedUrl` instead of `finalUrl` for `runFilesystemTarget ` function.
- Updated the test cases for `target=filesystem` in `upload.test.js`
- Updated the test cases for `upload-url-hash.test.js`

**SUPPORTING INFO**:
- This in relation to the ongoing discussion [here](https://github.com/GoogleChrome/lighthouse-ci/discussions/834#discussioncomment-4256769)
- #835 is part I of the implementation

### CODE REVIEW

**Comment by jaspritt-hora** [comment]
@connorjclark / @patrickhulce  - Please review the changes. I have made changes wrt `target=filesystem` only for now.

**Review by patrickhulce — COMMENTED** [review]
technically breaking but seems fine to me 🤷

File: packages/cli/test/upload-url-hash.test.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /**
- * @license Copyright 2019 Google Inc. All Rights Reserved.
```
Comment (connorjclark) [inline packages/cli/test/upload-url-hash.test.js:2]: please revert - as a policy we don't update these years

Comment (jaspritt-hora) [reply packages/cli/test/upload-url-hash.test.js]: @connorjclark - Done and I have squashed the commits as well

**Comment by connorjclark** [comment]
Thanks @jaspritt-hora!

**Pull request merged**